data-avro/src/main/java/com/linkedin/data/avro/DataTranslator.java

```diff
@@ -666,6 +645,19 @@ else if (fieldValue == null)
                 continue;
               }
             }
+
+            if ((fieldDataSchema.getDereferencedType() != DataSchema.Type.UNION) && fieldAvroSchema.getType() == Schema.Type.UNION) {
+              // Since optional field will be represented as union in Avro
+              // Need to extract the Avro type corresponding to the pegasus type from the Avro union
+              Map.Entry<String, Schema> fieldAvroEntry = findUnionMember(fieldDataSchema, fieldAvroSchema);
+              if (fieldAvroEntry == null)
+              {
+                _path.removeLast();
+                continue;
+              }
+              fieldAvroSchema = fieldAvroEntry.getValue();
+            }
+
             // Translate default value as null, depending on specified options
```

Comment:
Realizing that this comment makes no sense. I think you accidentally copied one of the comments from above in your last PR. You should fix this.
`// Translate default value as null, depending on specified options` -> `// Translate the pegasus field data to Avro data`